photon-ml/src/main/scala/com/linkedin/photon/ml/optimization/Optimizer.scala

```diff
@@ -14,252 +14,218 @@
     // Initialize the optimizer state if it's not being initialized yet, or if we don't need to reuse the existing
     // initial state for consistent convergence check across multiple runs.
     if (getInitialState.isEmpty || !isReusingPreviousInitialState) {
       setInitialState(getCurrentState)
     }
-    init(getCurrentState.get, data, objectiveFunction, initialCoefficients)
+    init(getCurrentState.get, objectiveFunction, initialCoefficients)
     do {
-      val updatedState = runOneIteration(data, objectiveFunction, getCurrentState.get)
+      val updatedState = runOneIteration(objectiveFunction, getCurrentState.get)
       setPreviousState(getCurrentState)
       setCurrentState(Some(updatedState))
     } while (!isDone)
     setConvergenceReason()
     val currentState = getCurrentState
     (currentState.get.coefficients, currentState.get.value)
   }
-
-  /**
-   * Solve the provided convex optimization problem.
-   * @param data The training data
-   * @param objectiveFunction The objective function to be optimized
-   * @return Optimized coefficients and the optimized objective function's value
-   */
-  def optimize(data: RDD[Datum], objectiveFunction: Function): (Vector[Double], Double) = {
-    optimize(Left(data), objectiveFunction)
-  }
-
-  /**
-   * Solve the provided convex optimization problem.
-   * @param data The training data
-   * @param objectiveFunction The objective function to be optimized
-   * @param initialCoefficients Initial coefficients
-   * @return Optimized coefficients and the optimized objective function's value
-   */
-  def optimize(data: RDD[Datum],
-               objectiveFunction: Function,
-               initialCoefficients: Vector[Double]): (Vector[Double], Double) = {
-    optimize(Left(data), objectiveFunction, initialCoefficients)
-  }
-
-  /**
-   * Solve the provided convex optimization problem.
-   * @param data The training data
-   * @param objectiveFunction The objective function to be optimized
-   * @return Optimized coefficients and the optimized objective function's value
-   */
-  def optimize(data: Iterable[Datum], objectiveFunction: Function): (Vector[Double], Double) = {
-    optimize(Right(data), objectiveFunction)
-  }
-
-  /**
-   * Solve the provided convex optimization problem.
-   * @param data The training data
-   * @param objectiveFunction The objective function to be optimized
-   * @param initialCoefficients Initial coefficients
-   * @return Optimized coefficients and the optimized objective function's value
-   */
-  def optimize(data: Iterable[Datum],
-               objectiveFunction: Function,
-               initialCoefficients: Vector[Double]): (Vector[Double], Double) = {
-    optimize(Right(data), objectiveFunction, initialCoefficients)
-  }
 }
+
+object Optimizer {
+  val DEFAULT_CONSTRAINT_MAP = None
+  val DEFAULT_TRACKING_STATE = true
+  val DEFAULT_REUSE_PREVIOUS_INIT_STATE = true
+}
```

Comment:
@joshvfleming 
Both the non-Scala-style constant names and missing newline are noted.